src/service/market/getPrice.ts

```diff
@@ -23,17 +28,39 @@ interface GetPriceReturn {
   prices: PriceDataByDate[] // list of price points
 }
 
-export default async function getPrice(params: GetPriceParams): Promise<GetPriceReturn> {
-  const { denom, interval, count } = params
-  const queryDatetimes = getQueryDatetimes(interval, Math.max(1, Math.min(1000, count)))
-  const prices = await getRepository(PriceEntity).find({
-    where: {
-      denom,
-      datetime: In(queryDatetimes)
-    },
-    order: { datetime: 'ASC' }
-  })
+type PriceByInterval = {
+  denom: string
+  price: number
+  datetime: string
+  time_div: number
+}
+
+async function getPriceAvg(params: GetPriceParams): Promise<PriceByInterval[]> {
+  const now = Date.now()
+  const msc = Math.max(MIN_DURATION, parseDuration(params.interval) || MIN_DURATION)
+  const intervalInSec = msc / 1000
+  const latestTimestamp = now - (now % msc)
+  const maxTimeStamp = now + msc * 2 // to make sure not to end up in zero segment of time diff
+  const minTimeStamp = latestTimestamp - msc * params.count
 
+  const subTimeQ = `trunc((((
+    DATE_PART('day', '${getQueryDateTime(maxTimeStamp)}'::timestamp - datetime::timestamp) * 24 
+    + DATE_PART('hour', '${getQueryDateTime(maxTimeStamp)}'::timestamp - datetime::timestamp)) * 60
+    + DATE_PART('minute', '${getQueryDateTime(maxTimeStamp)}'::timestamp - datetime::timestamp)) * 60 
+    + DATE_PART('second', '${getQueryDateTime(maxTimeStamp)}'::timestamp - datetime::timestamp))/(${intervalInSec}))`
+
+  const rawQ = `select denom, avg(price.price) as price, min(datetime) as datetime, ${subTimeQ} as time_div 
+    from price where denom = '${params.denom}' and datetime >= '${getQueryDateTime(minTimeStamp)}' 
+    group by denom, time_div order by time_div desc`
+
+  const prices: PriceByInterval[] = await getConnection().query(rawQ)
```

Comment:
If it is necessary to use raw query, you should use parameter instead of stitching strings. Variable must be escaped.

Reply:
Please capitalize SQL syntax words (as I updated on everywhere)